components/clp-py-utils/clp_py_utils/s3_utils.py

```diff
@@ -1,18 +1,177 @@
+        secret_access_key=credentials.secret_key,
+        session_token=credentials.token,
+    )
+
+
+def get_credential_env_vars(config: S3Config) -> Dict[str, str]:
+    """
+    Generates AWS credential environment variables for tasks.
+    :param config: S3Config or S3InputConfig from which to retrieve credentials.
+    :return: A [str, str] Dict which access key pair and session token if applicable.
+    :raise: ValueError if auth type is not supported
+    """
+    env_vars: Dict[str, str] = None
+    auth = config.aws_authentication
+
+    if "profile" == auth.type:
+        aws_credentials: S3Credentials = _get_session_credentials(auth.profile)
+        if aws_credentials is None:
+            raise ValueError(f"Failed to authenticate with profile {auth.profile}")
+
+        env_vars = {
+            "AWS_ACCESS_KEY_ID": aws_credentials.access_key_id,
+            "AWS_SECRET_ACCESS_KEY": aws_credentials.secret_access_key,
+        }
+        aws_session_token = aws_credentials.session_token
+        if aws_session_token is not None:
+            env_vars["AWS_SESSION_TOKEN"] = aws_session_token
+        return env_vars
+
+    elif "credentials" == auth.type:
+        credentials = auth.credentials
+        env_vars = {
+            "AWS_ACCESS_KEY_ID": credentials.access_key_id,
+            "AWS_SECRET_ACCESS_KEY": credentials.secret_access_key,
+        }
+        if credentials.session_token:
+            env_vars["AWS_SESSION_TOKEN"] = credentials.session_token
+        return env_vars
+
+    elif "env_vars" == auth.type:
+        # Environmental variables are already set
+        return {}
+    elif "ec2" == auth.type:
+        aws_credentials: S3Credentials = _get_session_credentials()
+        if aws_credentials is None:
+            raise ValueError(f"Failed to authenticate with EC2 metadata.")
+
+        env_vars = {
+            "AWS_ACCESS_KEY_ID": aws_credentials.access_key_id,
+            "AWS_SECRET_ACCESS_KEY": aws_credentials.secret_access_key,
+        }
+        if aws_credentials.session_token:
+            env_vars["AWS_SESSION_TOKEN"] = aws_credentials.session_token
+        return env_vars
+
+    else:
+        raise ValueError(f"Unsupported authentication type: {auth.type}")
+
+
+def get_container_authentication(
+    clp_config: CLPConfig, type: Literal["compression", "log_viewer", "query"]
+) -> Tuple[bool, List[str]]:
+    """
+    Generates Docker container authentication options for AWS S3 access based on the given type.
+    Handles authentication methods that require extra configuration (profile, env_vars).
+
+    :param clp_config: CLPConfig containing storage configurations
+    :param type: Type of calling container (compression, log_viewer, or query)
+    :return: Tuple of (whether aws config mount is needed, credential env_vars to set)
+    :raises: ValueError if environment variables are not set correctly or if type is invalid
+    """
+    if type not in ["compression", "log_viewer", "query"]:
```

Comment:
instead of defining your own types, I think you can reuse things like "COMPRESSION_WORKER_COMPONENT_NAME"